tests/callbacks_test.go

```diff
@@ -601,3 +601,174 @@ func (s *CallbacksSuite) TestNexusResetWorkflowWithCallback() {
+		RequestId:          uuid.NewString(),
+		Namespace:          s.Namespace().String(),
+		WorkflowId:         workflowID,
+		WorkflowType:       &commonpb.WorkflowType{Name: "blockingWorkflow"},
+		TaskQueue:          taskQueue,
+		Input:              nil,
+		WorkflowRunTimeout: durationpb.New(20 * time.Second),
+		Identity:           s.T().Name(),
+	}
+
+	startResponse1, err := s.FrontendClient().StartWorkflowExecution(ctx, request1)
+	s.NoError(err)
+
+	// Validate the workflow started, then terminate it
+	workflowExecution := &commonpb.WorkflowExecution{
+		WorkflowId: workflowID,
+		RunId:      startResponse1.RunId,
+	}
+	s.WaitForHistoryEvents(`
+			1 WorkflowExecutionStarted
+			2 WorkflowTaskScheduled
+			3 WorkflowTaskStarted
+			4 WorkflowTaskCompleted`,
+		s.GetHistoryFunc(s.Namespace().String(), workflowExecution),
+		5*time.Second,
+		10*time.Millisecond)
+
+	_, err = s.FrontendClient().TerminateWorkflowExecution(ctx, &workflowservice.TerminateWorkflowExecutionRequest{
+		Namespace:         s.Namespace().String(),
+		WorkflowExecution: workflowExecution,
+		Reason:            s.T().Name(),
+		Identity:          tv.WorkerIdentity(),
+	})
+	s.NoError(err)
+
+	s.EqualHistoryEvents(`
+			1 WorkflowExecutionStarted
+			2 WorkflowTaskScheduled
+			3 WorkflowTaskStarted
+			4 WorkflowTaskCompleted
+			5 WorkflowExecutionTerminated`,
+		s.GetHistoryFunc(s.Namespace().String(), workflowExecution)())
+
+	// 2. Start WF second time w/ callbacks (new run)
+	cbs := []*commonpb.Callback{
+		{Variant: &commonpb.Callback_Nexus_{Nexus: &commonpb.Callback_Nexus{Url: "http://" + callbackAddress + "/cb1"}}},
+	}
+
+	request2 := proto.Clone(request1).(*workflowservice.StartWorkflowExecutionRequest)
+	request2.RequestId = uuid.NewString()
+	request2.CompletionCallbacks = cbs
+
+	startResponse2, err := s.FrontendClient().StartWorkflowExecution(ctx, request2)
+	s.NoError(err)
+	s.True(startResponse2.Started)
+	s.NotEqual(workflowExecution.RunId, startResponse2.RunId)
+
+	s.WaitForHistoryEvents(`
+			1 WorkflowExecutionStarted
+			2 WorkflowTaskScheduled
+			3 WorkflowTaskStarted
+			4 WorkflowTaskCompleted`,
+		s.GetHistoryFunc(s.Namespace().String(), &commonpb.WorkflowExecution{WorkflowId: workflowID, RunId: startResponse2.RunId}),
+		5*time.Second,
+		10*time.Millisecond)
+
+	// 3. Reset workflow back to the first (terminated) run as base; must copy callbacks
+	resetWfResponse, err := sdkClient.ResetWorkflowExecution(ctx, &workflowservice.ResetWorkflowExecutionRequest{
+		Namespace:                 s.Namespace().String(),
+		WorkflowExecution:         workflowExecution, // base = first (terminated) run
+		Reason:                    s.T().Name(),
+		WorkflowTaskFinishEventId: 4,
+		RequestId:                 "test_id",
+	})
+	s.NoError(err)
+
+	// 4. Verify states instead of waiting for callback deliveries in this sequence
+	desc2, err := sdkClient.DescribeWorkflowExecution(ctx, workflowID, startResponse2.RunId)
```

Comment:
You should wait for the callbacks to be delivered here into the callback handler via the channels as we've done in other tests in this file. No need to describe the workflow.